server/gateway.go

```diff
@@ -3023,31 +3027,42 @@ func (c *client) gatewaySwitchAccountToSendAllSubs(e *insie, accName string) {
 
 // Keeps track of the routed reply to be used when/if application
 // sends back a message on the reply without the prefix.
-// Client lock held on entry. This is a server receiver because
-// we use a timer interval that is avail in Server.gateway object.
-func (s *Server) trackGWReply(c *client, reply []byte) {
+// This is a server receiver because we use a timer interval that
+// is avail in Server.gateway object.
+func (s *Server) trackGWReply(acc *Account, reply, routedReply []byte) {
+	acc.mu.Lock()
+	defer acc.mu.Unlock()
```

Comment:
Maybe I could unlock after adding to the map after line 3062.

Reply:
I like the way defer signals should be held throughout which I think is fine. Also prevents dangling locks so would keep as is.